### ISSUE TITLE
Regular segfaults

### ISSUE THREAD

**Issue by rhefner1** [influxdata/influxdb#12141]
I'm not sure exactly how to debug these segfaults, so I'm posting the details here to see if anyone has an idea of what's going on. I'm running influxdb (docker tag: 1.7.3-alpine) on Kubernetes with plenty of resources (500Mi CPU and 30Gi RAM, but no CPU or memory limits enforced). I'm running two instances but only 1 is affected by this issue, which makes me suspect it has something specific to the data on the instance.

I do suspect that this is a memory-related issue but despite reading many posts, I still don't fully understand how InfluxDB uses virtual memory and how that works in Kubernetes. At the time of writing, the resident size (RES) in top is 27.8G and virtual (VIRT) is 47.3. Kubernetes reports that the pod is taking 41.8G. The node itself only has 46.4G of addressable memory.

It has panicked 23 times over the last 7 days (from Elasticsearch). Sometimes there are multiple in a row and sometimes they are standalone:

```
February 25th 2019, 14:20:28.560	default-influxdb-kube	fatal error: fault
February 25th 2019, 14:16:05.585	default-influxdb-kube	fatal error: fault
February 25th 2019, 14:10:08.789	default-influxdb-kube	fatal error: fault
February 24th 2019, 08:58:02.796	default-influxdb-kube	fatal error: fault
February 24th 2019, 07:04:34.781	default-influxdb-kube	fatal error: fault
February 24th 2019, 07:02:04.783	default-influxdb-kube	fatal error: fault
February 24th 2019, 06:43:05.788	default-influxdb-kube	fatal error: fault
February 22nd 2019, 16:33:31.252	default-influxdb-kube	fatal error: fault
February 22nd 2019, 04:26:05.783	default-influxdb-kube	fatal error: fault
February 22nd 2019, 03:14:04.781	default-influxdb-kube	fatal error: fault
February 21st 2019, 12:37:05.789	default-influxdb-kube	fatal error: fault
February 21st 2019, 12:31:08.810	default-influxdb-kube	fatal error: fault
February 20th 2019, 21:23:16.782	default-influxdb-kube	fatal error: fault
February 20th 2019, 20:58:34.782	default-influxdb-kube	fatal error: fault
February 20th 2019, 20:48:05.783	default-influxdb-kube	fatal error: fault
February 20th 2019, 11:27:18.423	default-influxdb-kube	fatal error: fault
February 20th 2019, 11:08:18.406	default-influxdb-kube	fatal error: fault
February 20th 2019, 11:00:14.412	default-influxdb-kube	fatal error: fault
February 19th 2019, 14:56:14.381	default-influxdb-kube	fatal error: fault
February 19th 2019, 14:50:24.371	default-influxdb-kube	fatal error: fault
February 18th 2019, 23:47:15.371	default-influxdb-kube	fatal error: fault
February 18th 2019, 23:44:05.371	default-influxdb-kube	fatal error: fault
February 18th 2019, 23:40:18.370	default-influxdb-kube	fatal error: fault
```

Here is an example of a full panic traceback: https://gist.github.com/rhefner1/ddd0a18f769c475fedec7c2727d3b30a#file-panic_1-txt

Here is a list of some of the queries that were logged right before each of the crashes:
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'default-gitlab' AND time > now() - 5m GROUP BY time(1m) fill(previous)`
- `SELECT median(value), max(value) FROM k8s.\"default\".\"cpu/usage_rate\" WHERE (container_name =~ /^prod-manager-api$/) AND time >= now() - 12h GROUP BY time(30s) fill(none)`
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'kube-system-heapster-sink' AND time > now() - 5m GROUP BY time(1m) fill(none)`
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'default-influxdb-kube' AND time > now() - 5m GROUP BY time(1m) fill(none)`
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'kube-system-heapster-sink' AND time > now() - 5m GROUP BY time(1m) fill(none)`
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'default-influxdb-kube' AND time > now() - 5m GROUP BY time(1m) fill(none)`
- `SELECT count(value) FROM k8s.\"default\".uptime WHERE container_name = 'default-influxdb-kube' AND time > now() - 5m GROUP BY time(1m) fill(none)`

`influx_inspect` returns no errors:

```
 bash-4.4# influx_inspect verify -dir /var/lib/influxdb
/var/lib/influxdb/data/_internal/monitor/145/000000006-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/148/000000007-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/149/000000007-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/150/000000006-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/151/000000006-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/152/000000005-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/153/000000006-000000002.tsm: healthy
/var/lib/influxdb/data/_internal/monitor/154/000000002-000000002.tsm: healthy
/var/lib/influxdb/data/k8s/default/137/000010287-000000002.tsm: healthy
/var/lib/influxdb/data/k8s/default/137/000010287-000000003.tsm: healthy
/var/lib/influxdb/data/k8s/default/137/000010287-000000004.tsm: healthy
/var/lib/influxdb/data/k8s/default/146/000009017-000000005.tsm: healthy
/var/lib/influxdb/data/k8s/default/146/000009017-000000006.tsm: healthy
/var/lib/influxdb/data/k8s/default/146/000009017-000000007.tsm: healthy
/var/lib/influxdb/data/k8s/default/146/000009017-000000008.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000512-000000005.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000640-000000004.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000771-000000004.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000803-000000003.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000804-000000001.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000805-000000001.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000806-000000001.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000807-000000001.tsm: healthy
/var/lib/influxdb/data/k8s/default/155/000000808-000000001.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/138/000000114-000000003.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/147/000000132-000000002.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/156/000000008-000000002.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/156/000000009-000000001.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/156/000000010-000000001.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/156/000000012-000000001.tsm: healthy
/var/lib/influxdb/data/ksm/autogen/156/000000013-000000001.tsm: healthy
Broken Blocks: 0 / 26856227, in 80.697461144s
```

---

So, do these segfaults look like what would happen if the node were out of memory and influx was trying to allocate more? Or is there another issue?

**Comment by nefethael** [comment]
I'm having some segfaults too, resolved by following commandline:
`INFLUXDB_MONITOR_STORE_ENABLED=false influxdb`

Not sure we have same issue, just a hint ...

**Comment by rhefner1** [comment]
@nefethael I increased the memory of the node and disabled the monitor store and the segfaults seems to have stopped (only 3 in the last 3 days as opposed to 20). So one of those things must've fixed it. Thanks!

**Comment by e-dard** [comment]
This issue should be fixed in v1.7.4.
